test/e2e/kubevirt.go

```diff
@@ -1420,4 +1426,131 @@ passwd:
+					return networkStatus.Default
+				})
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(primaryUDNNetworkStatuses).To(HaveLen(1))
+				primaryUDNNetworkStatus = primaryUDNNetworkStatuses[0]
+			})
+
+			By("Wait NetworkManager readiness")
+			Eventually(func() error {
+				_, err := virtLauncherCommand("systemctl is-active NetworkManager")
+				return err
+			}).
+				WithTimeout(5 * time.Second).
+				WithPolling(time.Second).
+				Should(Succeed())
+
+			By("Reconfigure primary UDN interface to use dhcp/nd for ipv4 and ipv6")
+			_, err := virtLauncherCommand(kubevirt.GenerateAddressDiscoveryConfigurationCommand("ovn-udn1"))
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+		It("should configure IPv4 and IPv6 using DHCP and NDP", func() {
+
+			dnsService, err := fr.ClientSet.CoreV1().Services(config.Kubernetes.DNSServiceNamespace).
+				Get(context.Background(), config.Kubernetes.DNSServiceName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			if isIPv4Supported() {
+				expectedIP, err := util.MatchIPStringFamily(false /*ipv4*/, primaryUDNNetworkStatus.IPs)
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedDNS, err := util.MatchIPStringFamily(false /*ipv4*/, dnsService.Spec.ClusterIPs)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, cidr, err := net.ParseCIDR(cidrIPv4)
+				Expect(err).ToNot(HaveOccurred())
+				expectedGatway := util.GetNodeGatewayIfAddr(cidr).IP.String()
+
+				Eventually(primaryUDNValueForConnection).
+					WithArguments("DHCP4.OPTION").
+					WithTimeout(10 * time.Second).
+					WithPolling(time.Second).
+					Should(ContainElements(
+						"host_name = vm1",
+						fmt.Sprintf("ip_address = %s", expectedIP),
+						fmt.Sprintf("domain_name_servers = %s", expectedDNS),
+						fmt.Sprintf("routers = %s", expectedGatway),
+						fmt.Sprintf("interface_mtu = 1300"),
+						fmt.Sprintf("requested_interface_mtu = 1"),
+					))
+				Expect(primaryUDNValueForConnection("IP4.ADDRESS")).To(ConsistOf(expectedIP + "/24"))
+				Expect(primaryUDNValueForConnection("IP4.GATEWAY")).To(ConsistOf(expectedGatway))
+				Expect(primaryUDNValueForConnection("IP4.DNS")).To(ConsistOf(expectedDNS))
+				Expect(primaryUDNValueForDevice("GENERAL.MTU")).To(ConsistOf("1300"))
+			}
+
+			if isIPv6Supported() {
+				expectedIP, err := util.MatchIPStringFamily(true /*ipv6*/, primaryUDNNetworkStatus.IPs)
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(primaryUDNValueFor).
+					WithArguments("connection", "DHCP6.OPTION").
+					WithTimeout(10 * time.Second).
+					WithPolling(time.Second).
+					Should(ContainElements(
+						"fqdn_fqdn = vm1",
+						fmt.Sprintf("ip6_address = %s", expectedIP),
+					))
+				Expect(primaryUDNValueForConnection("IP6.ADDRESS")).To(SatisfyAll(HaveLen(2), ContainElements(expectedIP+"/128")))
+				Expect(primaryUDNValueForConnection("IP6.GATEWAY")).To(ConsistOf(WithTransform(func(ipv6 string) bool {
+					return netip.MustParseAddr(ipv6).IsLinkLocalUnicast()
```

Comment:
It's quite involve since it means we have to read ovn-nbctl to get LRP's mac and calculate the LinkLocal address, don't think is worth it, we are not going to test OVN IPv6 capabilities.

Maybe follow up

Reply:
I had thought it would be possible to extract it from either the node's annotations, or compute it from the config. 

Yeah, an hypothetical follow-up is good enough for me.